workalendar/europe/ireland.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+from datetime import date, timedelta
+from workalendar.core import WesternCalendar, ChristianMixin
+from workalendar.core import MON
+
+
+class Ireland(WesternCalendar, ChristianMixin):
+    "Republic of Ireland"
+
+    include_easter_monday = True
+    include_boxing_day = True
+    boxing_day_label = "St. Stephen's Day"
+
+    shift_new_years_day = True
+
+
+    def get_june_holiday(self, year):
+        return (
+            Ireland.get_nth_weekday_in_month(year, 6, MON),
+            "June Holiday"
+        )
+
+    def get_august_holiday(self, year):
+        return (
+            Ireland.get_nth_weekday_in_month(year, 8, MON),
+            "August Holiday"
+        )
+
+    def get_variable_days(self, year):
+        days = super(Ireland, self).get_variable_days(year)
+
+        # St Patrick's day
+        st_patrick = date(year, 3, 17)
+        days.append((st_patrick, "Saint Patrick's Day"))
+        if st_patrick.weekday() in self.get_weekend_days():
+            days.append((
+                self.find_following_working_day(st_patrick),
+                "Saint Patrick substitute"))
+
+        # Whit Monday
+        if year <= 1973:
```

Comment:
Maybe you'd need to add a test to check that, before 1974, this day was added to the holidays